contracts/oracle/src/lib.rs

```diff
@@ -149,18 +149,25 @@ impl Contract {
             reference_hash: None,
         };
 
-        self.used_identities.insert(&external_id);
```

Comment:
this was moved to the callback to avoid potential reentrancy attack

Reply:
No, it should not be moved to the callback, otherwise someone can do 2 claims in the same block. It must be done here.

Reply:
DONE